sys/net/application_layer/emcute/emcute.c

```diff
@@ -351,8 +351,14 @@ int emcute_pub(emcute_topic_t *topic, const void *data, size_t len,
     tbuf[pos++] = flags;
     byteorder_htobebufs(&tbuf[pos], topic->id);
     pos += 2;
-    byteorder_htobebufs(&tbuf[pos], id_next);
-    waitonid = id_next++;
+    /* for QOS0 we must set the MsgId to zero */
+    if (flags & MQTTSN_QOS_MASK) {
```

Comment:
I guess this one needs to be checked for QoS -1 as well.

Reply:
same here: does not make any practical diference, as QoS -1 is not supported by emcute. But handling this more explicitly does indeed not hurt -> fixed.